### PR TITLE
[avro-c] Fix building avro-c in Linux

### PR DESCRIPTION
**Pull request by mispecto**
This PR fixes Avro for C compilation in Linux (issue #10339 ).

Changes:

 - Fix finding debug versions of `liblzma`, `libsnappy` and `jansson`
 - Link with standard C++ library when Snappy is added as reference (it's written in C++).

### CODE REVIEW

File: ports/avro-c/CONTROL [hunk 1]
```diff
@@ -1,5 +1,5 @@
 Source: avro-c
-Version: 1.9.2
+Version: 1.9.2-1
 Supports: !(uwp|linux|osx)
```
Comment (kreuzerkrieg) [inline ports/avro-c/CONTROL]: Looks like you need to remove `linux` here

**Comment by kreuzerkrieg** [comment]
looks like it should build on OSX too now, am I wrong?

**Comment by spektom** [comment]
> 
> 
> looks like it should build on OSX too now, am I wrong?

Probably, but I don't have Mac in my possession to check this :)

**Comment by kreuzerkrieg** [comment]
Me neither, let the CI test it :)

**Comment by spektom** [comment]
> 
> 
> Me neither, let the CI test it :)

BTW, do you know why the CI fails for every platform? I can't find a meaningful log that explains the failure..

**Comment by kreuzerkrieg** [comment]
> > Me neither, let the CI test it :)
> 
> BTW, do you know why the CI fails for every platform? I can't find a meaningful log that explains the failure..

Looks like they have problems with the CI. Again.
Usually when they resolve the problem they re-run the CI. Just lets wait for @LilyWangL to take care of it

**Comment by LilyWangL** [comment]
> > > Me neither, let the CI test it :)
> > 
> > 
> > BTW, do you know why the CI fails for every platform? I can't find a meaningful log that explains the failure..
> 
> Looks like they have problems with the CI. Again.
> Usually when they resolve the problem they re-run the CI. Just lets wait for @LilyWangL to take care of it

Waiting for the PR #10508 merged, this PR will fix the CI test.

**Comment by LilyWangL** [comment]
@spektom Thanks for your PR, please modify ci.baseline.txt to account for the Linux failures. Remove the following code in VCPKG_PATH/scripts/ci.baseline.txt:
```
avro-c:x64-linux=fail
```

**Comment by spektom** [comment]
> @spektom Thanks for your PR, please modify ci.baseline.txt to account for the Linux failures. Remove the following code in VCPKG_PATH/scripts/ci.baseline.txt:
> 
> ```
> avro-c:x64-linux=fail
> ```

Done.

**Review by LilyWangL — APPROVED** [review]

**Comment by kreuzerkrieg** [comment]
bump

**Comment by kreuzerkrieg** [comment]
bump

**Review by PhoebeHui — APPROVED** [review]

**Pull request merged**